tests/subsys/portability/cmsis_rtos_v2/src/event_flags.c

```diff
@@ -216,16 +222,6 @@ void test_event_flags_isr(void)
 
 	zassert_true(osEventFlagsDelete(evt_id) == osOK,
 		     "EventFlagsDelete failed");
-}
-ZTEST(cmsis_event_flags, test_event_flags)
-{
-	/*
-	 *These tests are order-dependent.
-	 *They have to be executed in order.
-	 *So put these tests in one ZTEST.
-	 */
-	test_event_flags_no_wait_timeout();
-	test_event_flags_signalled();
-	test_event_flags_isr();
+	osEventFlagsDelete(evt_id);
```

Comment:
you are now free-ing an already freed event flag.

Reply:
> IMO, this should be done in the setup/teardown functions to keep the changes to the test logic minimal.

I have put it in the before function. Please have a look again. Thanks. @stephanosio